### ISSUE TITLE
Removing a table, after bluring and focusing again an empty editor log warnigns

### ISSUE THREAD

**Issue by mlewand** [ckeditor/ckeditor-dev#874]
## Are you reporting a feature request or a bug?

Bug

 ## [Check if the issue is already reported](http://docs.ckeditor.com/#!/guide/dev_issues_readme-section-avoid-duplicates)

*Put all reference links here…*

 ## Provide detailed reproduction steps (if any)

1. Open CKEditor sample.
2. Open browser's console.
1. Clear the editor's content.
3. Insert table with nested table.
4. Select outer table.
5. Blur the editor.
6. Refocus the editor, but doesn't change the selection.
7. Press <kbd>Delete</kbd>.

 ### Expected result

Table is removed.

 ### Actual result

Table is removed, but there are warnings in the console:

> [CKEDITOR] Error code: range-startcontainer.

> Error code: range-endcontainer.

 ## Other details

This issue surfaced after fix in #867.

**Comment by mlewand** [comment]
tl;dr: it should be fixed by #873.

I did some debugging on this, here are the notes:

This exception is caused due to invalid return value of [`range.getBoundaryNodes()`](https://github.com/ckeditor/ckeditor-dev/blob/141fd23fd6ea1da8edd7f2996c34fe181ecb0732/plugins/tableselection/plugin.js#L908).

For for selection like:

```html
<html>
<head></head>
<body>^</body>
</html>
```

it returns startNode = `<head>`, endNode = `<head>`, which is a nonsense.

The exception comes from [`rng.moveToElementEditablePosition( startNode );`](https://github.com/ckeditor/ckeditor-dev/blob/141fd23fd6ea1da8edd7f2996c34fe181ecb0732/plugins/tableselection/plugin.js#L949) where startNode is a `head` element.


My assumption is that it is [due to wrong implementation of `getNextSourceNode` / `getPrevSourceNode`](https://dev.ckeditor.com/ticket/12191#comment:2).

**Comment by mlewand** [comment]
**I have added unit tests** to the `t/874` branch.
